src/planning/planner.ts

```diff
@@ -246,9 +246,7 @@ function plan(
         if (
             isStackOverflowExeption(error)
         ) {
-            if (context.plan) {
-                circularDependencyToException(context.plan.rootRequest);
-            }
+            circularDependencyToException(context.plan.rootRequest);
```

Comment:
I decided that the condition was not necessary given that the planner creates the context and sets the plan and there is no way to intercept it.